contracts/protocore/SelfProtocore.sol

```diff
@@ -109,4 +109,38 @@ contract SelfProtocore is MasterCopyNonUpgradable, GenesisSelfProtocore, Protoco
             }
         }
     }
+
+    /**
+     * @notice Registers a vote for a link specified by vote message hash.
+     *
+     * @dev Satisfies \pre and \post conditions of Protocore::registerVoteInternal().
+     *
+     * \pre If a link (specified by `_voteMessageHash`) length is equal to one
+     *      epoch length (finalisation link) the inclusion principle should
+     *      be satisfied:
+     *      - current block number is strictly less than
+     *        target block height + epoch length
+     */
+    function registerVote(
```

Comment:
Missing documentation for params and also the `\post` condition

Reply:
Maybe remove the `@dev` 

The \pre and \post condition can be like this, an example.

* \pre A link must exist in the link mapping for the given `_voteMessageHash`.
~* \pre The current block number must be greater than the target checkpoint of the link.~
* \pre The current block number must be less than the sum of the target block number of link and epoch length.
* \pre Satisfy the \pre conditions of `Protocore::registerVoteInternal()`.
*
* \post Satisfy the \post conditions of `Protocore::registerVoteInternal()`.